src/NuGet.Core/NuGet.Configuration/PackageNamespaces/PackageNamespacesConfiguration.cs

```diff
@@ -35,14 +38,22 @@ public class PackageNamespacesConfiguration
         /// <exception cref="ArgumentException"> if <paramref name="term"/> is null, empty, or whitespace only.</exception>
         public IReadOnlyList<string> GetConfiguredPackageSources(string term)
         {
-            return SearchTree.Value?.GetConfiguredPackageSources(term);
+            IReadOnlyList<string> sources = SearchTree.Value?.GetConfiguredPackageSources(term);
+
+            if (!_namespaceModeStrategy.TryValidate(term, sources, out string errormessage))
```

Comment:
I'm wondering if throwing an exception is the best API design here.

Reply:
I had the same concern Nikolche. Referred [doc](https://learning.oreilly.com/library/view/framework-design-guidelines/9780321545671/chapter07.html) to recap best practices and noticed below guidance.

**DO** `report execution failures by throwing exceptions.` - `If a member cannot successfully do what it is designed to do, it should be considered an execution failure, and an exception should be thrown`.

Given that the intent of `GetConfiguredPackageSources` method is to `get package sources with matching prefix from package namespaces section. ` which is different from validating namespace mode behavior. Hence, I pushed a new commit tweaking API a bit. Please review and let me know the feedback.